### PR TITLE
Small workflow updates

### PR DESCRIPTION
**Pull request by ausias-armesto**
## Summary by CodeRabbit

- **New Features**
	- Introduced a `publish` job to enhance the CI/CD pipeline.

- **Improvements**
	- Enhanced flexibility in Node.js version management and Docker image tagging using variables.
	- Updated team reviewer assignments for code reviews.

- **Bug Fixes**
	- Improved reliability and performance in retrieving pull request versions.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -9,29 +9,20 @@ concurrency:
...
 jobs:
-    build-and-publish:
-        name: Build and Push
+    build:
+        name: Build
         runs-on: self-hosted-hoprnet-small
-
+        strategy:
+            matrix:
+                node-version: [20.x, 21.x, 22.x]
```
Comment (esterlus) [inline .github/workflows/build.yaml]: only v20 and v22 should be checked as of now. see https://nodejs.org for current LTS and stable versions

**Review by esterlus — CHANGES_REQUESTED** [review]
Can you add some info about Deployment to the README?

File: .github/workflows/build.yaml (hunk 1 above)
Comment (ausias-armesto) [reply .github/workflows/build.yaml]: Thanks, already removed in all repositories

**Review by esterlus — APPROVED** [review]
Looks good

**Pull request merged**